src/main/java/racingcar/domain/Cars.java

```diff
@@ -1,39 +1,73 @@
 package racingcar.domain;
 
-import racingcar.dto.CarsResponseDto;
 import racingcar.utils.RandomGenerator;
 
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.NoSuchElementException;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
+import static java.util.stream.Collectors.counting;
+import static java.util.stream.Collectors.groupingBy;
+
 public class Cars {
+    private static final String SAME_NAME_ERROR_MSG_FORMAT = "[ERROR] 중복된 이름이 있습니다 : %s";
+    private static final String COMMA_AND_BLANK = ", ";
+
     private final List<Car> cars;
 
-    public Cars(List<Car> cars) {
-        this.cars = cars;
+    private Cars(List<Car> cars) {
+        validateSameName(cars);
+        this.cars = new ArrayList<>(cars);
     }
 
-    public List<Car> getCars() {
-        return Collections.unmodifiableList(cars);
+    public static Cars createCars(List<Car> cars) {
+        return createCars(cars);
+    }
+
+    public static Cars createCarsByNames(List<String> carNames) {
+        return new Cars(carNames.stream()
+                .map(Car::new)
+                .collect(Collectors.toList()));
```

Comment:
```suggestion
        return carNames.stream()
                .map(Car::new)
                .collect(Collectors.collectingAndThen(Collectors.toList(), this::createCars));
```

collectingAndThen을 활용할 수 있겠네요 :)

Reply:
앗 맞네요 아직 자바 8 문법이 숨쉬듯 자연스럽진 않습니다!!
연습연습